### ISSUE TITLE
Trying to export data on layers that have arrays in searchUrl will make the loading flag goes in loop

### ISSUE THREAD

**Issue by MV88** [geosolutions-it/MapStore2#9970]
## Description

download fails with layers with multiple urls

## How to reproduce

-  try to import this map
[download.json](https://github.com/geosolutions-it/MapStore2/files/14223536/download.json)
- to download the layer in shapefile

Another way to reporoduce the issues is to:
- edit wms catalog by adding more urls in the url field with the comma
![image](https://github.com/geosolutions-it/MapStore2/assets/11991428/566df502-14c1-4783-a81f-704724ebb47d)
- then add a layer 
- try to export the shapefile for a layer

*Expected Result*

it works 
*Current Result*

![image](https://github.com/geosolutions-it/MapStore2/assets/11991428/02cbea36-b594-4766-865b-edd92e214374)

- [x] Not browser related

<details><summary> <b>Browser info</b> </summary>

(use this site: <a href="https://www.whatsmybrowser.org/">https://www.whatsmybrowser.org/</a> for non expert users)

| Browser Affected | Version |
|---|---|
|Internet Explorer| |
|Edge| |
|Chrome| |
|Firefox| |
|Safari| |
</details>

## Other useful information

Basically we need to avoid to let the user to insert a comma separated list into the wms service catalog. For backwards compatibility we need to perform a check when MAP_CONFIG_LOADED is done in order to parse correctly the layer config, i would do it inside the normalizeLayer function
basically the url property should not be an array but only one value. this will fix this issue.

**Comment by offtherailz** [comment]
@MV88 I noticed that the url array is actually documented and supported on MapStore doc. I agree to separate the concept of domain sharding. 
The domain aliases option is present in catalog advanced configuration but it doesn't seems to be working, so the only solution now supported is the comma-saparated list. 

So, if you want to fix this bug simply, we should do a pre-check on export to include the possibility it is an array.

If you don't want to make it possible to be an array we have to handle : 
- `domainAliases` option to be passed to the layer when added to the map
- `domainAliases` property documented and effectively used by layer openlayers/leaflet/cesium implementations
- `deprecate` in doc the usage of url array/list (doc/jsdoc)
- Optional (IMHO) If you don't want the users to insert a list of comma-separated URLs, you probably should make sure that the catalog input is a valid URL (this means absolute URL, or absolte / relative path ), but this regards the validation of the URL in general.

Please sync with @tdipisa to check what to do, if fix the bug or do this stuff. 
In any case, domain aliases configuration do not work and it have to be fixed.

**Comment by tdipisa** [comment]
@offtherailz the **_Domain aliases_** functionalities seems perfectly working to me: I've created [this test map](https://dev-mapstore.geosolutionsgroup.com/mapstore/#/viewer/48757).

The point here is that probably when we have introduced the support for _Domain aliases_ we have missed to properly handle retro compatibility for layers previously saved with multiple URLs separated by comma. 
Therefore the idea for me would be to make MS able to transparently convert the layer state in the new expected format for _Domain aliases_ when the map is loaded.

**Comment by offtherailz** [comment]
Sorry for `domainAliases` I did a wrong test. For the array, it is the current implementation, so we have to use it

**Comment by tdipisa** [comment]
@offtherailz ok I understood the problem. The format on which the URLs are saved in the state is always the one using an array as before even using the New UI by defining multiple Domain aliases. For exmaple:

```
                "url": [
                    "https://maps4.geosolutionsgroup.com/geoserver/wms",
                    "https://maps1.geosolutionsgroup.com/geoserver/wms",
                    "https://maps2.geosolutionsgroup.com/geoserver/wms",
                    "https://maps3.geosolutionsgroup.com/geoserver/wms",
                    "https://maps4.geosolutionsgroup.com/geoserver/wms",
                    "https://maps5.geosolutionsgroup.com/geoserver/wms"
                ],
```

there is effectively a bug for which this format is not the one expected by the Export tool, for which we have the following error:

```
mapstore2.js?b0226e63a3df897834bd:2 Error in epic "startFeatureExportDownload". Original error: TypeError: Parameter 'url' must be a string, not object
    at i.parse (mapstore2.js?b0226e63a3df897834bd:2:4597132)
    at Object.b [as parse] (mapstore2.js?b0226e63a3df897834bd:2:4597059)
    at k (mapstore2.js?b0226e63a3df897834bd:2:960019)
    at M (mapstore2.js?b0226e63a3df897834bd:2:961249)
    at Lt (layerDownload.b0226e63a3df897834bd.chunk.js:1:23933)
    at t.project (layerDownload.b0226e63a3df897834bd.chunk.js:1:29257)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4312554)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4270310)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4280363)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4270310)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4336079)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
```

And by the Style selector tab for which we have instead this error:

```
mapstore2.js?b0226e63a3df897834bd:2 Error in epic "toggleStyleEditorEpic". Original error: TypeError: n.url.split is not a function
    at t.project (styleEditor.b0226e63a3df897834bd.chunk.js:1:4387)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4312554)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4270310)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4270310)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4280363)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4270310)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t._next (mapstore2.js?b0226e63a3df897834bd:2:4336079)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4181888)
    at t.next (mapstore2.js?b0226e63a3df897834bd:2:4178762)
    at mapstore2.js?b0226e63a3df897834bd:2:4148552
    at mapstore2.js?b0226e63a3df897834bd:2:4148538
```
